src/core/types/scope_spec_identifier.rs

```diff
@@ -1,312 +0,0 @@
-use crate::core::error::ContractError;
-use crate::core::types::serialized_enum::SerializedEnum;
-use schemars::JsonSchema;
-use serde::{Deserialize, Serialize};
-
-use crate::util::{
-    aliases::AssetResult,
-    scope_address_utils::{
-        scope_spec_address_to_scope_spec_uuid, scope_spec_uuid_to_scope_spec_address,
-    },
-    traits::ResultExtensions,
-};
-
-const UUID_NAME: &str = "uuid";
-const ADDRESS_NAME: &str = "address";
-
-/// An enum containing interchangeable values that can be used to define a Provenance Blockchain Metadata Scope Specification.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-pub enum ScopeSpecIdentifier {
-    /// A uuid v4 represented by a string.
-    Uuid(String),
```

Comment:
I kept kicking myself for naming this `Uuid` because each "type" has to be globally unique when deserializing values in the `SerializedEnum` space, so this is the best possible way to undo that mistake 🍡